marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/impl/CreatedOnFacetHandler.java

```diff
@@ -21,33 +21,115 @@
 import com.marklogic.hub.central.entities.search.Constants;
 import com.marklogic.hub.central.entities.search.FacetHandler;
 import com.marklogic.hub.central.entities.search.models.DocSearchQueryInfo;
+import com.marklogic.hub.central.exceptions.DataHubException;
 
 import java.time.LocalDate;
 import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAdjusters;
+import java.util.HashMap;
+import java.util.Map;
 
 public class CreatedOnFacetHandler implements FacetHandler {
 
-    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
-    private static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter
-            .ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+    public static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ssZ");
+
+    int zoneOffset = 0;
```

Comment:
These can't be class fields, because EntitySearchManager is storing one instance of it in a static map that is used for all threads. So we'll get thread-safety issues on this and possibly incorrect results. 

A good rule of thumb is that if you have a set of fields that are associated with one another, toss them into a class. So instead of making these class fields of CreatedOnFacetHandler, make an e.g. CreatedOnFacetInputs class and create a new instance of that when buildQuery is called, and then pass it to each case method. This avoids thread-safety issues and reduces the amount of state (which is generally good - the more state, the more problems there are).